dotnet/Cucumber.Messages.Specs/MessagesSpec.cs

```diff
@@ -1,31 +1,68 @@
 using Xunit;
 ﻿using System.IO;
-using static Io.Cucumber.Messages.PickleStepArgument.Types;
+using Io.Cucumber.Messages.Types;
+using System.Text.Json;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text.Json.Serialization;
+using System;
+using System.ComponentModel;
 
 namespace Io.Cucumber.Messages.Specs
 {
     public class MessagesSpec
     {
         [Fact]
-        public void SerializesPickeDocString()
+        public void SerializesAStepDefinitionToNDJSONCorrectly()
         {
-            var pickleDocSring = new PickleDocString
+            var stepDefinitionNDJSON = @"{""stepDefinition"":{""id"":""0"",""pattern"":{""source"":""I have {int} cukes in my belly"",""type"":""CUCUMBER_EXPRESSION""},""sourceReference"":{""location"":{""line"":3},""uri"":""samples/minimal/minimal.feature.ts""}}}
+";          
+            var soureReference = new SourceReference("samples/minimal/minimal.feature.ts",
+                null, null, new Location(3, null));
+            var stepDefPattern = new StepDefinitionPattern("I have {int} cukes in my belly", StepDefinitionPatternType.CUCUMBER_EXPRESSION);
+            var stepDefinition = new StepDefinition("0", stepDefPattern, soureReference);
+            var env = new Envelope(null, null, null, null, null, null, null, null, stepDefinition, null, null, null, null, null, null, null, null);
+
+            var jsonOptions = new JsonSerializerOptions();
+            jsonOptions.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
+            jsonOptions.Converters.Add(new CustomEnumConverter<StepDefinitionPatternType>());
+            jsonOptions.DefaultIgnoreCondition = System.Text.Json.Serialization.JsonIgnoreCondition.WhenWritingNull;
```

Comment:
These json options are non trivial and will have to be replicated by anyone who wants to serialize messages. So I think it would be prudent to provide a serializer and deserializer class as part of the messages library. This should hopefully prevent any mistakes from taking hold.

You may also want to look at some of the other utility classes that come with messages. 

https://github.com/cucumber/messages/tree/main/java/src/main/java/io/cucumber/messages

When comparing do note that unlike many other languages, Java does not have an out of the box json serializer  On the other hand, the libraries Java does have update quite frequently. To avoid this cascading down from messages, the `MessageToNdjsonWriter` is constructed with a `Serializer` interface instead. This allows us to keep the serializer library at the edge of the dependency tree.

Reply:
Thanks for the quick feedback. I'll take a look at whether the utility classes make sense in the dotnet ecosystem. 
I'll double check with Gaspar about the .net version (5 v 8). 
And I'll port over the Java test suite (or something similar).

Reply:
Completed. Please take a look at what was added to the support classes to make serialization easier. It aligns with how the Java implementation works.